errors.go

```diff
@@ -0,0 +1,53 @@
+package zoholab
+
+import (
+	"fmt"
+
+	jsoniter "github.com/json-iterator/go"
+
+	"github.com/pkg/errors"
+)
+
+// drop-in replacement for encoding/json
+var json = jsoniter.ConfigCompatibleWithStandardLibrary
+
+// ZohoError is the error that the zoho api returns.
+type ZohoError struct {
+	Response *Response `json:"response"`
+}
+
+// Response is part of the ZohoError response Model.
+type Response struct {
+	URI           string        `json:"uri"`
+	Action        string        `json:"action"`
+	ResponseError ResponseError `json:"error"`
+}
+
+// ResponseError is part of the ZohoError response Model.
+type ResponseError struct {
+	Code    int64  `json:"code"`
+	Message string `json:"message"`
+}
+
+// Error stringifies the ServerException Error.
+func (apiError ZohoError) Error() string {
+	return fmt.Sprintf("Uri: %s, Action: %s, ErrorCode: %d, ErrorMessage: %s",
+		apiError.Response.URI,
+		apiError.Response.Action,
+		apiError.Response.ResponseError.Code,
+		apiError.Response.ResponseError.Message)
+}
+
+// FillApiError helper function used to fill the ZohoError model.
+func FillApiError(respBody []byte) error {
```

Comment:
Du könntest stattdessen auch einfach die SetError function von resty nutzen

Reply:
SetError funktioniert nicht da der error einfach nur in den response body zurückgegeben wird.

Reply:
Genau dafür ist SetError.
Ist der Status dabei dann trotzdem 200 oder wie?

Reply:
yep